homeassistant/components/automation/trace.py

```diff
@@ -1,26 +1,17 @@
 """Trace support for automation."""
 from __future__ import annotations
 
-from collections import OrderedDict
 from contextlib import contextmanager
 import datetime as dt
-from datetime import timedelta
 from itertools import count
-import logging
-from typing import Any, Awaitable, Callable, Deque
+from typing import Any, Deque
 
-from homeassistant.core import Context, HomeAssistant, callback
-from homeassistant.helpers.json import JSONEncoder as HAJSONEncoder
+from homeassistant.components.trace.const import DATA_TRACE, STORED_TRACES
```

Comment:
I don't think that we should allow other integrations to be aware of `hass.data[DATA_TRACE]`. We should abstract that.

Reply:
Fixed in #48276